test/cluster/test_view_building_coordinator.py

```diff
@@ -539,3 +539,31 @@ async def test_view_building_failure(manager: ManagerClient):
 
         await wait_for_view(cql, 'mv_cf_view', node_count)
         await check_view_contents(cql, ks, "tab", "mv_cf_view")
+
+# Reproduces scylladb/scylladb#25912
+@pytest.mark.asyncio
+async def test_mv_rf_change(manager: ManagerClient):
+    # This tests created 2 nodes per DC (node1/dc1, node2/dc1, node1/dc2, node2/dc2)
+    # and the initial tablet value in the keyspace is 1.
+    # So when the base table and the view are created, initially there are:
+    # - 2 tablets on node1/dc1
+    # - 0 tablets on node2/dc1
+    # - 2 tablets on node1/dc2
+    # - 0 tablets on node2/dc2
+    # Because the base table and the view aren't colocated, load balancer creates 2 migration for the same table:
+    # - in dc1: tablet of base table node1 -> node2
+    # - in dc2: tablet of base table node1 -> node2
+    # Since these are 2 migrations of different tablets but in the same table,
+    # one of them is overwritten by the second one and this leads to # Reproduces scylladb/scylladb#25912
+    servers = await manager.servers_add(4, cmdline=cmdline_loggers, property_file=[
+    {"dc": "dc1", "rack": "myrack1"},
+    {"dc": "dc1", "rack": "myrack2"},
+    {"dc": "dc2", "rack": "myrack1"},
+    {"dc": "dc2", "rack": "myrack2"}
+])
+
+    cql = manager.get_cql()
+    await cql.run_async("CREATE KEYSPACE IF NOT EXISTS ks WITH replication = {'class': 'NetworkTopologyStrategy', 'dc1': 1, 'dc2': 1} AND tablets = {'initial': 1}")
+    await cql.run_async("CREATE TABLE ks.base (pk int, ck int, PRIMARY KEY (pk, ck))")
+    await cql.run_async("CREATE MATERIALIZED VIEW ks.mv AS SELECT pk, ck FROM ks.base WHERE ck IS NOT NULL PRIMARY KEY (ck, pk)")
+    await wait_for_view(cql, "mv", 4)
```

Comment:
Can this race with load balancing migrations, and the view gets built before migrations are finished, thus making the test pass even if the bug is present?

Reply:
You probably want to add `manager.api.quiesce_topology()` to ensure that load balancing is finished.